crates/typst-syntax/src/lexer.rs

```diff
@@ -16,12 +16,35 @@ pub(super) struct Lexer<'s> {
     mode: LexMode,
     /// Whether the last token contained a newline.
     newline: bool,
-    /// The state held by raw line lexing.
-    raw: Vec<(SyntaxKind, usize)>,
+    /// The state held by raw line lexing. This is emptied into an inner SyntaxNode before
+    /// returning from [`lex_past_trivia`].
+    raw: Vec<SyntaxNode>,
     /// An error for the last token.
+    ///
+    /// This is present to increase convenience when returning an error by avoiding the
+    /// need to manually return an error from most lexer functions.
```

Comment:
Like mentioned in another comment, it feels somewhat unfortunate that we still keep `raw` and `error` even though they are not really necessary with the new design.

Reply:
I could that, but if we did, then we would likely make most of the other lexer functions return a Result type or other SyntaxNode item. I could be convinced of this, but it would be more effort.

Reply:
I would have thought just a SyntaxNode. Not sure how it affects performance, but overall I think it would reap the rewards of the whole refactor.

Reply:
I don't think it's going to help that much at this stage. I tried converting `Lexer::error()` into creating a  syntax node and it just adds a ton of noise everywhere. 
I also did a draft of moving the raw Vec into the raw function and having it return a SyntaxNode, but it has annoyances due to how `Lexer::code` is currently called when checking for the newline mode.

Reply:
I do think the change for math field accesses is a win. Curous what your thougts are there.